File: java/002.java
class Problem002 {
	public static void main(String[] args) {
		int sum = 0;
		int a = 1, b = 2;
		while (a < 4000000) {
			if (a % 2 == 0) {
				sum += a;
			}			

			int c = a + b;
			a = b;
			b = c;
		}
		System.out.println(sum);
	}
}

File: exs/002.exs
fib = Stream.unfold({0, 1}, fn {a, b} -> {a, {b, a + b}} end)
even? = &(rem(&1, 2) == 0)
IO.puts fib |> Enum.take_while(&(&1 < 4_000_000)) |> Enum.filter(even?) |> Enum.sum
